src/NServiceBus.AcceptanceTesting/Support/TaskExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace NServiceBus.AcceptanceTesting.Support
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    static class TaskExtensions
+    {
+        public static Task Timebox(this IEnumerable<Task> tasks, TimeSpan timeoutAfter, string messageWhenTimeboxReached)
+        {
+            var taskCompletionSource = new TaskCompletionSource<object>();
+            var tokenSource = new CancellationTokenSource(timeoutAfter);
+            var registration = tokenSource.Token.Register(s =>
+            {
+                var tcs = (TaskCompletionSource<object>)s;
+                tcs.TrySetException(new Exception(messageWhenTimeboxReached));
+            }, taskCompletionSource);
+            {
+                Task.WhenAll(tasks)
+                    .ContinueWith((t, s) =>
+                    {
+                        var state = (Tuple<TaskCompletionSource<object>, CancellationTokenSource, CancellationTokenRegistration>)s;
+                        var source = state.Item2;
+                        var reg = state.Item3;
+                        var tcs = state.Item1;
+
+                        if (t.IsFaulted && t.Exception != null)
+                        {
+                            tcs.TrySetException(t.Exception.InnerException);
+                        }
+
+                        if (t.IsCanceled)
+                        {
+                            tcs.TrySetCanceled();
+                        }
+
+                        if (t.IsCompleted)
+                        {
+                            tcs.TrySetResult(null);
+                        }
+
+                        reg.Dispose();
```

Comment:
It could but then I need two tuples :( and I wanted to dispose it together with the source